src/controller/helpers/zclFrameConverter.ts

```diff
@@ -1,19 +1,26 @@
-import {ZclFrame} from '../../zcl';
+import {ZclFrame, Utils as ZclUtils} from '../../zcl';
+import Debug from "debug";
 
 interface KeyValue {[s: string]: number | string}
 
-function attributeKeyValue(frame: ZclFrame): KeyValue {
+function attributeKeyValue(frame: ZclFrame, manufacturerID?: number): KeyValue {
     const payload: KeyValue = {};
+    let cluster = frame.Cluster;
+
+    if (manufacturerID) {
+        Debug.log(`Attempting to resolve manufacturer specific (${manufacturerID}) cluster attributes.`);
+        cluster = ZclUtils.getCluster(frame.Cluster.ID, manufacturerID);
+    }
 
     for (const item of frame.Payload) {
         try {
-            const attribute = frame.Cluster.getAttribute(item.attrId);
+            const attribute = cluster.getAttribute(item.attrId);
             payload[attribute.name] = item.attrData;
+            Debug.log(`Attr: ${item.attrId} (${attribute.name}) -> ${item.attrData}}`);
```

Comment:
I propose to not add any debug logging here, it will pollute the log files a lot.